app/routes.py

```diff
@@ -1,2 +1,186 @@
+
+@goals_bp.route("", methods=["GET"])
+def get_goals():
+    """Defines an endpoint that gets all goals, optionally presented by a query pram of descending\
+        or ascending"""
+    query_sort = request.args.get("sort")
+    goals_response = []
+
+    if query_sort == "asc":
+        goals = Goal.query.order_by(Goal.title)
+    elif query_sort == "desc":
+        goals = Goal.query.order_by(Goal.title.desc())
+    else:
+        goals = Goal.query.all()
+        
+    for goal in goals:
+        goals_response.append(goal.to_dict())
+
+    return make_response(jsonify(goals_response), 200)
+
+@goals_bp.route("/<goal_id>", methods=["GET"])
+def get_one_goal(goal_id):
+    """Defines an endoint that returns all goals"""
+    goal = Goal.query.get_or_404(goal_id)
+    return make_response(jsonify({"goal": goal.to_dict()}), 200)
+
+@goals_bp.route("", methods=["POST"])
+def create_a_valid_goal():
+    """Defines an endpoint that creates a valid goal with all required attributes"""
+    request_body = request.get_json()
+
+    if "title" not in request_body:
+        return make_response(jsonify({"details": "Invalid data"}), 400)
+    
+    new_goal = Goal(
+        title = request_body["title"]
+        )
+    add_one_database(new_goal)
+    return make_response(jsonify({"goal": new_goal.to_dict()}), 201)
+
+@goals_bp.route("/<goal_id>", methods=["DELETE"])
+def delete_a_goal(goal_id):
+    """Defines an endpoint to delete a goal"""
+    goal = Goal.query.get_or_404(goal_id)
+    delete_one_database(goal)
+    return make_response(jsonify({"details": f'Goal {goal_id} "{goal.title}" successfully deleted'}), 200)
+
+@goals_bp.route("/<goal_id>", methods=["PUT"])
+def update_a_goal(goal_id):
+    """Defines an endpoint that updates a goal with new information"""
+    goal = Goal.query.get_or_404(goal_id)
+    request_body = request.get_json()
+    goal.title = request_body["title"]
+    db.session.commit
+    return make_response(jsonify({"goal": goal.to_dict()}), 200)
+
+@goals_bp.route("/<goal_id>/tasks", methods=["POST"])
+def post_tasks_for_goal(goal_id):
+    """Defines an endpoint that associates a task with a goal"""
+    request_body = request.get_json()
+    goal = Goal.query.get_or_404(goal_id)
+    for each_task in request_body["task_ids"]:
+        each_task = Task.query.get(each_task)
+        each_task.fk_goal_id = goal.goal_id
+    return make_response(jsonify({"id": goal.goal_id, "task_ids": request_body["task_ids"]}), 200)
+
+@goals_bp.route("/<goal_id>/tasks", methods=["GET"])
+def get_tasks_for_goal(goal_id):
+    """Defines an endpoint that returns all tasks associated with a goal """
+    goal = Goal.query.get_or_404(goal_id)
+    return make_response(goal.to_dict_with_tasks(goal_id), 200)
+
+def sanitize_date(request_body):
+    """Verifies that complete_at in correct datetime format"""
+    if request_body["completed_at"]:
+        try:
+            val = request_body["completed_at"]
+            type(val) == datetime
```

Comment:
I like the idea here! To complete this, it would probably be best to using a parsing function from [datetime](https://docs.python.org/3/library/datetime.html) or [dateutil](https://dateutil.readthedocs.io/en/stable/).